### ISSUE TITLE
[BUG] The number of selected items doesn't match the number of rows in the Excel export file.

### ISSUE THREAD

**Issue by Dave-Biddle-Soak** [Maatwebsite/Laravel-Nova-Excel#121]
### Prerequisites

* [X] Able to reproduce the behaviour outside of your code, the problem is isolated to Laravel Excel.
* [X] Checked that your issue isn't already filed.
* [X] Checked if no PR was submitted that fixes this problem.

### Versions

* PHP version: 7.3.28
* Laravel version: 8.40.0
* Nova version: 3.24.0
* Package version:  Laravel Nova Excel: 3.1.30 (From Changelog). Laravel Excel: Not installed

### Description

When using "Select All", or manually selecting items to export, then running the excel export action, the number of selected items does not match the number of exported rows.

When using "Select All Matching", the number of exported rows does match the total number of items shown.

### Steps to Reproduce
1. Visit a Nova Resource page.
2. Use the Select All option to select all items on the page, or manually select some items.
3. Run the Excel export action to download the Excel export file.

**Expected behavior:**

The number of rows in the exported `.xlsx` should match the number of selected items.

**Actual behavior:** 

The number of exported rows is less than the number of selected items.

### Additional Information

I traced the count of items to be less than the number of selected items in the result of:
```
// Maatwebsite\LaravelNovaExcel\Actions\ExportToExcel:89
$query = $this->request->toExportQuery();

// debug code
logger($query->count()); // less than the number of selected items
```

**Comment by patrickbrouwers** [comment]
Are you using a custom query in your resource?

**Comment by Dave-Biddle-Soak** [comment]
Hi @patrickbrouwers thanks for your reply. Yes we are using a custom query in our resource. Guess this is likely to be the cause of the issue? If so, I wonder if there is a way to make the Excel export work with a resource which using a custom query?

Any help would be much appreciated! Thanks in advance.

**Comment by patrickbrouwers** [comment]
Yes that's most likely the issue then. It should work with a custom query, but probably it something specific to the one you wrote.

**Comment by Dave-Biddle-Soak** [comment]
Hi @patrickbrouwers, firstly apologies for not replying sooner. 

We are using a number of `join` conditions on our query, so I was wondering whether you know if the following code uses the `indexQuery` method of a Resource at all?

```
// Maatwebsite\LaravelNovaExcel\Actions\ExportToExcel:81 
public function handleRequest(ActionRequest $request)
{
    // ...
    $this->request  = ExportActionRequestFactory::make($request);

    $query = $this->request->toExportQuery();
```

If it does, then I would expect our join conditions to be respected, so perhaps the code isn't using our `indexQuery` method?

Any help would be greatly appreciated.

**Comment by patrickbrouwers** [comment]
We use the `toSelectedResourceQuery` which is something that Nova offers. Try checking what the raw sql is that is being executed.

**Comment by Dave-Biddle-Soak** [comment]
Hi @patrickbrouwers thanks for your suggestion, I did some debugging inside the `toSelectedResourceQuery` method and am happy to report that I found the cause of the issue, which turned out to be a bug in our code after all. So my apologies for asserting that "the problem is isolated to Laravel Excel" while creating this issue!

The cause of the issue was that the `id` column in the `indexQuery` method of our resource was actually set to the `id` column of a different model to that being shown in the Nova dashboard and exported in the XLS file. This meant that the following code in your package was adding an `IN` clause to the query, which contained the id's of a different resource:

```php
   // vendor/maatwebsite/laravel-nova-excel/src/Requests/ExportResourceActionRequest.php
    public function toExportQuery()
    {
        return $this->toSelectedResourceQuery()->when(!$this->forAllMatchingResources(), function ($query) {
            $query->whereKey(explode(',', $this->resources));
        });
    }

```
The fix was to amend the code in the `indexQuery` method of our resource, to use the `id` column of the model defined in the `$model` property of the resource. This meant that the query was filtering by the correct ids and the correct items were exported in the XLS export.

```php
    /**
     * The model the resource corresponds to.
     *
     * @var string
     */
    public static $model = OurResource::class;

    public static function indexQuery(NovaRequest $request, $query)
    {
        return $query->select([
                'our_resource.id',
                // ...
```

Closing this now as the issue is resolved. Many thanks for your help.
